File: src/main/java/com/itmo/microservices/users/service/UserService.java
package com.itmo.microservices.users.service;

import com.itmo.microservices.users.repository.UserRepository;
import lombok.AllArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@AllArgsConstructor
public class UserService {

    private final UserRepository userRepository;
}

File: src/main/kotlin/com/itmo/microservices/demo/users/api/service/UserService.kt
package com.itmo.microservices.demo.users.api.service

import com.itmo.microservices.demo.users.api.model.AppUserModel
import com.itmo.microservices.demo.users.api.model.RegistrationRequest
import org.springframework.security.core.userdetails.UserDetails
import java.util.*

interface UserService {
    fun findUser(username: String): AppUserModel?
    fun registerUser(request: RegistrationRequest): AppUserModel
    fun getAccountData(requester: UserDetails?, userId: UUID): AppUserModel
}